torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -981,6 +982,32 @@ def test_average_parameters(self):
                 for p in model.parameters():
                     self.assertEqual(p.data, torch.ones_like(p.data) * -rank)
 
+        @unittest.skipIf(
+            BACKEND != "nccl" and BACKEND != "gloo",
+            "MPI backend does not support creating subgroups on CUDA devices",
+        )
+        @skip_if_lt_x_gpu(2)
+        def test_periodic_model_averager(self):
+            rank = dist.get_rank()
+            rank_to_GPU = self._init_multigpu_helper()
+            device_id = rank_to_GPU[rank][0]
+            world_size = dist.get_world_size()
+
+            model = nn.Linear(1, 5, bias=False).cuda(device_id)
+            param = next(model.parameters())
+            tensor = torch.ones_like(param.data) * rank
```

Comment:
should this be just `torch.ones_like(param)`?

Reply:
`parameter` is not a normal tensor, but has  both `.data` and `.grad` fields.